src/pupil_labs/realtime_api/models.py

```diff
@@ -279,3 +283,105 @@ def direct_eyes_sensor(self) -> T.Optional[Sensor]:
                 sensor=Sensor.Name.EYES.value, conn_type=Sensor.Connection.DIRECT.value
             ),
         )
+
+
+@dataclasses.dataclass
+class TemplateItem:
+    widget_type: str
+    choices: T.Optional[T.List[str]] = None
+    help_text: T.Optional[str] = None
+    id: T.Optional[uuid.UUID] = None
+    input_type: str = dataclasses.field(default="any")
+    required: bool = dataclasses.field(default=False)
+    title: T.Optional[str] = None
+
+    def __post_init__(self):
+        widget_type_enum = {
+            "TEXT",
+            "PARAGRAPH",
+            "RADIO_LIST",
+            "CHECKBOX_LIST",
+            "SECTION_HEADER",
+            "PAGE_BREAK",
+        }
+        input_type_enum = {"any", "integer", "float"}
+
+        if self.widget_type not in widget_type_enum:
+            raise ValueError(
+                f"Invalid value for widget_type: {self.widget_type}. Must be one of {widget_type_enum}."
+            )
+
+        if self.input_type not in input_type_enum:
+            raise ValueError(
+                f"Invalid value for input_type: {self.input_type}. Must be one of {input_type_enum}."
+            )
+
+    @classmethod
+    def fromdict(cls, data: T.Dict) -> "TemplateItem":
+        return cls(
+            widget_type=data["widget_type"],
+            choices=data.get("choices"),
+            help_text=data.get("help_text"),
+            id=uuid.UUID(data["id"]) if data.get("id") else None,
+            input_type=data.get("input_type", "any"),
+            required=data.get("required", False),
+            title=data.get("title"),
+        )
+
+
+@dataclasses.dataclass
+class Template:
+    archived_at: T.Optional[datetime.datetime] = None
+    archived_by_user_id: T.Optional[uuid.UUID] = None
+    created_at: T.Optional[datetime.datetime] = None
+    created_by_user_id: T.Optional[uuid.UUID] = None
+    description: T.Optional[str] = None
+    id: T.Optional[uuid.UUID] = None
+    is_default_template: bool = dataclasses.field(
+        default=False, metadata={"readonly": True}
+    )
+    items: T.List[TemplateItem] = dataclasses.field(default_factory=list)
+    label_ids: T.List[uuid.UUID] = dataclasses.field(
+        default_factory=list, metadata={"readonly": True}
+    )
+    name: str = ""
+    published_at: T.Optional[datetime.datetime] = None
+    recording_ids: T.List[uuid.UUID] = dataclasses.field(
+        default_factory=list, metadata={"readonly": True}
+    )
+    recording_name_format: T.Optional[str] = None
+    updated_at: T.Optional[datetime.datetime] = None
+
+    @classmethod
+    def fromdict(cls, data: T.Dict) -> "Template":
```

Comment:
Something to consider in the future is something like pydantic and get all the validation/conversion for free, eg:

```python
from uuid import UUID
from datetime import datetime
from typing import Literal
from pydantic.dataclasses import dataclass



@dataclass
class Template:
    id: UUID
    created_at: datetime
    items: list[TemplateItem]

WidgetType = Literal["TEXT", "PARAGRAPH", "RADIO_LIST", "CHECKBOX_LIST", "SECTION_HEADER", "PAGE_BREAK"]

@dataclass
class TemplateItem:
    widget_type: WidgetType
    title: str

json_data = {
    "id": "99ae566d-5988-4e47-8147-297d9ac989a4",
    "created_at": "2022-01-01T01:01:01",
    "items": [
        {"widget_type": "TEXT", "title": "a question"},
    ],
}

Template(**json_data)
# Template(id=UUID('99ae566d-5988-4e47-8147-297d9ac989a4'), created_at=datetime.datetime(2022, 1, 1, 1, 1, 1), items=[TemplateItem(widget_type='TEXT', title='a question')])
```